### ISSUE TITLE
Gtk3: Gtk.SelectionData.set() is broken in python bindings

### ISSUE THREAD

**Issue by genodeftest** [jaap-karssenberg/zim-desktop-wiki#390]
`Gtk.SelectionData.set()` cannot set selection data for dragging within file browser or attachment browser. See also: #385
Upstream bug reports:
* https://gitlab.gnome.org/GNOME/pygobject/issues/104 has a workaround which can be found at https://github.com/virtuald/pygi-treeview-dnd. See also: exaile/exaile#200
* https://gitlab.gnome.org/GNOME/pygobject/issues/90

**Comment by jaap-karssenberg** [comment]
Trying to get this to work using pygi-treeview-dnd but currently blocked by this issue: https://github.com/virtuald/pygi-treeview-dnd/issues/4

**Comment by virtuald** [comment]
Weird, I'll take a look later tonight.

**Comment by jaap-karssenberg** [comment]
Recently it seems this is now working. Keeping bug open for monitoring the situation, but apparently something changes upstream.

Gtk / python-gi versions of working situation:
- libgtk-3-0/bionic,now 3.22.30-1ubuntu1
- python3-gi/bionic,now 3.26.1-2

**Comment by murattepegoz** [comment]
Folks, dragging files from attachment panel into a wiki page is also not working for me. The problem started with version 0.70. I am using zim on mac with homebrew. Let me know if I need to create a new issue for this.
Here is the snapshot summarizing the problem.
![image](https://user-images.githubusercontent.com/3308687/56845551-f017cf80-6877-11e9-944f-b39124534156.png)

**Comment by jaap-karssenberg** [comment]
This is a separate issue please report in a new ticket.

Regards,

Jaap

Op za 27 apr. 2019 08:08 schreef murattepegoz <notifications@github.com>:

**Comment by jhe-** [comment]
The issue is still happening for me.

I'm on:
- Zim 0.71.0
- libgtk-3-0/bionic,now 3.22.30-1ubuntu2
- python3-gi/bionic,now 3.26.1-2

**Comment by jhe-** [comment]
#827 is a hacky but minimal workaround that fixes drag-and-drop in the index pane in my case.

**Comment by arigit** [comment]
Issue still present on 0.71.1, running from source tree, MacOS 
Drag & drop from Finder to a zim page results in nothing. The same action on the same maching on 0.69.1 (also running from source tree) works perfectly.

Running zim in debug mode (python3 zim.py --debug) shows no messages at all when dropping the file.

**Comment by dalcolmo** [comment]
Issue still present on 0.71.1, running on Xubuntu 18.04.
dragging a page to a place under another page does nothing. Using the right-click move command works.
I also keep getting crash report windows, but as far as I can tell no crash happened.

**Comment by ragat69** [comment]
Reordering pages in the Index pane is broken on Linux Mint 19.02. Moving pages around does nothing, they stay where they are.
zim Version : 0.72

**Comment by hartwork** [comment]
If this affects 0.72: Can we adjust the title to say `zim >=0.70` rather than `zim 0.70`?

**Comment by jaap-karssenberg** [comment]
Committed a workaround for this issue in the development branch just now, please test if it solves the issue

**Comment by chihchun** [comment]
Tested 8190f499[1] on bionic, it did not solve the issue.

libgtk-3-0:amd64	3.22.30-1ubuntu4
python3-gi	3.26.1-2ubuntu1

[1] Workaround for drag-and-drop issue #390 · zim-desktop-wiki/zim-desktop-wiki@8190f49 - https://github.com/zim-desktop-wiki/zim-desktop-wiki/commit/8190f499eabb52b48002270d656764885ad982b3

**Comment by marenkov** [comment]
The same problem with Zim 0.73.0
Ubuntu 16.04

**Comment by elzinus** [comment]
Same issue... ``zim -D`` gives the following information.

```
DEBUG: Drag data requested, we have internal path "00 INBOX:01 NOTITIES:0002 a2 Vim"
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/zim/plugins/pageindex/__init__.py", line 512, in do_drag_data_received
    assert len(names) == 1
AssertionError
```

I would love to test the solution #340 ; how can I do this?

**Comment by jaap-karssenberg** [comment]
Please test again with the "develop" branch - added additional logging to diagnose what is happening here

**Comment by galenthurber** [comment]
same here
monitoring

**Comment by jaap-karssenberg** [comment]
Just released 0.73.2 which has more logging for this issue. Also there is a
"view debug log" menu option now under tools. Please include debug
output so I can better diagnose the issue.

**Comment by BorgeJorge** [comment]
I don't know if this is related, but 0.73.2 still has a problem that's been true for all versions after 0.65 (Windows) that I've tried, which is that when I drag and drop a file from Windows Explorer onto a Zim page, the link always gets put at the bottom of the page regardless of where I drop it or where the current cursor position is. It's why I still use 0.65.

**Comment by jaap-karssenberg** [comment]
No that is a separate issue. Afraid that one is also an upstream GTK issue
I can't easily solve in the application.

Op vr 24 jul. 2020 21:26 schreef BorgeJorge <notifications@github.com>:

**Comment by elzinus** [comment]
Tested again, see result below.

`
(zim:17063): Gtk-CRITICAL **: 13:24:06.468: gtk_tree_drag_source_drag_data_delete: assertion 'iface->drag_data_delete != NULL' failed
`
zim 0.73.2

**Comment by 777vollbio** [comment]
Windows 10: If a file (pdf, png, xslx, ... ) is dragged from the Explorer into the application using the mouse, it is always inserted at the top (not bottom like comment @BorgeJorge #24 Jul 2020 ) of the page and not where the mouse pointer is pointing at. Ctrl+c/p doesn't work. Zim 0.74.1

**Comment by osman1963** [comment]
Moving page with drag and drop inside the idnex doesn't work.
Windows 10, Zim 0.7.4.2

**Comment by lnz999** [comment]
> Windows 10: If a file (pdf, png, xslx, ... ) is dragged from the Explorer into the application using the mouse, it is always inserted at the top (not bottom like comment @BorgeJorge #24 Jul 2020 ) of the page and not where the mouse pointer is pointing at. Ctrl+c/p doesn't work. Zim 0.74.1

Would like to confirm this. File links, when dropping files on a page, used to be inserted at the bottom  of the page like @BorgeJorge #states for earlier versions. Now they are inserted at the top of the page, changing also the formatting of the header. Also Ctrl+z does not restore the formatting. Using 0.74.2 on Windows 10.

**Comment by SBCV** [comment]
Same issue for Ubuntu 18.04 and zim 0.74.3. Guess I'll stick with zim 0.69 for now - where drag and drop still works.

**Comment by ghost** [comment]
Hello, I am using Zim 0.75.1 on Windows 10, and this problem is present as well.

**Comment by layerswing** [comment]
Fresh installed Ubuntu 22.04 zim 0.74.3 the drag and drop issue is still present.
Any solution to fix it?

**Comment by pedrib** [comment]
Using Debian, upgraded 0.73.5 to 0.75.1 and now drag and drop doesn't work any more! But unlike people commenting here, 0.73.5 was working fine with drag and drop for me.

**Comment by osman1963** [comment]
A setting in one of the plugins is responsible for this, can't remember which. It can be changed in the plugin to allow for drag and drop.

**Comment by bigpresh** [comment]
[for dragging pages in the index view around, at least...]

> A setting in one of the plugins is responsible for this, can't remember which. It can be changed in the plugin to allow for drag and drop.

Looks to be the "Page Index" plugin...

Edit -> Preferences -> Plugins tab -> "Page Index" plugin -> Configure :+1: 

![image](https://user-images.githubusercontent.com/54107/225868614-e7ac32a1-c630-4216-ae99-cae0ba83ba1e.png)

**Comment by layerswing** [comment]
Thank you [bigpresh](https://github.com/bigpresh),
I can confirm, that this is the solution for this feature. It works on Ubuntu 22.04 Zim 0.75.1.
@bigpresh +1

**Comment by pedrib** [comment]
@bigpresh also works for me, thank you!

**Comment by smaudet** [comment]
@bigpresh any chance we could get this default to true? This was the old behavior...

I am stuck using the old pre 0.70 builds for the most part, due to instability of the new GTK versions. It has made me want to fork Zim several times, I mostly don't have time to do so.

**Comment by prahal** [comment]
@smaudet you want to bug https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1699 which is about the index pane drag-and-drop.
This bug is about issues with drag and dropping files into zim (from a file manager for one).
There was an out-of-topic discussion in the middle about the index pane drag-and-drop, but it is best to keep it in its thread that is #1699 
I would also prefer a confirmation dialog (that could eventually be disabled by a setting) on the index pane page move than the current change of the default to no drag-and-drop.

Note that this change was triggered by https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1443 (which points to a few valid bugs that should probably be fixed: that is in non-editable mode zim still allows to move pages around and there is a bug when moving a parent).

**Comment by dalcolmo** [comment]
I am on Zim 0.75.2 on Kubuntu 5.15.0-105-generic.

Drag-and-drop from File manager (Dolphin) to the attachments pane used to work in 0.72 but now in 0.75.2 it only works if the attachment folder already exists.
